src/XRP/network-clients/web-socket-network-client.ts

```diff
@@ -0,0 +1,155 @@
+      } else {
+        this.handleErrorMessage(
+          'Received unhandlable message: ' + (event.data as string),
+        )
+      }
+    })
+
+    this.socket.addEventListener('close', (event) => {
+      this.handleErrorMessage(
+        'Web socket disconnected, ' + (event.reason as string),
+      )
+    })
+
+    this.socket.addEventListener('error', (event) => {
+      this.handleErrorMessage('Error: ' + (event as string))
+    })
+  }
+
+  /**
+   * Properly handles incoming transactions from the web socket.
+   *
+   * @param data The web socket response received from the web socket.
+   */
+  private handleTransaction = (data: WebSocketResponse) => {
+    const dataTransaction = data as WebSocketTransactionResponse
+    const destinationAccount = dataTransaction.transaction.Destination
+    const callback = this.accountCallbacks.get(destinationAccount)
+    if (callback) {
+      callback(dataTransaction)
+    } else {
+      throw new XrpError(
+        XrpErrorType.Unknown,
+        'Received a transaction for an account that has not been subscribed to: ' +
+          destinationAccount,
+      )
+    }
+  }
+
+  /**
+   * Sends an API request over the web socket connection.
+   *
+   * @param request The object to send over the web socket.
+   * @returns The API response from the web socket.
+   */
+  private async sendApiRequest(
+    request: WebSocketRequestOptions,
+  ): Promise<WebSocketStatusResponse> {
+    while (this.socket.readyState === 0) {
+      await sleep(5)
+    }
+    if (this.socket.readyState !== 1) {
+      throw new XrpError(XrpErrorType.Unknown, 'Socket is closed/closing')
+    }
+    this.socket.send(JSON.stringify(request))
+
+    this.waiting.set(request.id, undefined)
+    while (this.waiting.get(request.id) === undefined) {
+      await sleep(5)
+    }
+    const response = this.waiting.get(request.id)
+    this.waiting.delete(request.id)
+
+    return response as WebSocketStatusResponse
+  }
+
+  /**
+   * Subscribes to incoming transactions to a given account.
+   * @see https://xrpl.org/monitor-incoming-payments-with-websocket.html
+   *
+   * @param id The ID used for the subscription.
+   * @param callback The function called whenever a new transaction is received.
+   * @param account The account from which to subscribe to incoming transactions, encoded as a classic address.
+   * @returns The response from the websocket confirming the subscription.
+   */
+  public async subscribeToAccount(
+    id: string,
+    callback: (data: WebSocketTransactionResponse) => void,
+    account: string,
+  ): Promise<WebSocketStatusResponse> {
+    this.accountCallbacks.set(account, callback)
+    const options = {
+      id,
+      command: 'subscribe',
+      accounts: [account],
+    }
+    const response = await this.sendApiRequest(options)
+    if (response.status !== 'success') {
+      throw new XrpError(
+        XrpErrorType.Unknown,
+        'Subscription request for ' + account + ' failed',
```

Comment:
I wonder why a subscription request might fail... do you think there is any other information potentially lurking in the response that might be useful to pass to the user to help them fix their issue?